File: app/src/main/java/padam/keystroke_recognition/PressureTestActivity.java
package padam.keystroke_recognition;

import android.support.v7.app.AppCompatActivity;
import android.os.Bundle;
import android.text.Layout;
import android.view.MotionEvent;
import android.view.View;
import android.widget.Button;
import android.widget.TextView;

import butterknife.BindView;
import butterknife.ButterKnife;
import butterknife.OnClick;
import butterknife.OnTouch;

import static android.icu.lang.UCharacter.GraphemeClusterBreak.L;

public class PressureTestActivity extends AppCompatActivity {

    @BindView(R.id.activity_pressure_test) View pressureTest;
    @BindView(R.id.displayPressure) TextView displayPressureTextView;
    @BindView(R.id.displaySize) TextView displaySizeTextView;
    @BindView(R.id.displayMaxPressure) TextView displayMaxPressureTextView;
    @BindView(R.id.displayMaxSize) TextView displayMaxSizeTextView;
    @BindView(R.id.resetButton) Button resetButton;

    private float maxPressure = -1;
    private float maxSize = -1;

    @Override
    protected void onCreate(Bundle savedInstanceState) {
        super.onCreate(savedInstanceState);
        setContentView(R.layout.activity_pressure_test);
        ButterKnife.bind(this);
    }

    @OnTouch(R.id.activity_pressure_test)
    public boolean pressureTestClick(View v, MotionEvent motionEvent){
        float pressure = motionEvent.getPressure();
        float size = motionEvent.getSize();
        displayPressureTextView.setText("Pressure: "+String.valueOf(pressure));
        displaySizeTextView.setText("Size: "+String.valueOf(size));
        maxValues(pressure, size);
        return true;
    }

    @OnClick(R.id.resetButton)
    public void resetButtonClick(){
        displayPressureTextView.setText("");
        displaySizeTextView.setText("");
        displayMaxPressureTextView.setText("");
        displayMaxSizeTextView.setText("");
    }

    private void maxValues(float pressure, float size){
        if (maxPressure < pressure){
            maxPressure = pressure;
            displayMaxPressureTextView.setText("Maximum Pressure: "+String.valueOf(maxPressure));
        }
        if(maxSize < size){
            maxSize = size;
            displayMaxSizeTextView.setText("Maximum Size: "+String.valueOf(maxSize));
        }
    }
}

File: app/src/main/java/padam/keystroke_recognition/PressureTestActivity.kt
package padam.keystroke_recognition

import android.os.Bundle
import android.support.v7.app.AppCompatActivity
import android.view.MotionEvent
import kotlinx.android.synthetic.main.activity_pressure_test.*

class PressureTestActivity : AppCompatActivity() {

    var maxPressure = -1f
    var maxSize = -1f

    override fun onCreate(savedInstanceState: Bundle?) {
        super.onCreate(savedInstanceState)
        setContentView(R.layout.activity_pressure_test)
        activity_pressure_test.setOnTouchListener { _, event -> pressureTestClick(event) }
        resetButton.setOnClickListener { _ -> resetFields() }
    }

    private fun pressureTestClick(event: MotionEvent): Boolean {
        displayPressure.text = "Pressure: " + maxPressure.toString()
        displaySize.text = "Size: " + maxSize.toString()
        maxValues(event.pressure, event.size)
        return true
    }

    private fun resetFields() {
        displayPressure.text = ""
        displaySize.text = ""
        displayMaxPressure.text = ""
        displayMaxSize.text = ""
        maxPressure = -1f
        maxSize = -1f
    }

    private fun maxValues(pressure: Float, size: Float) {
        if (maxPressure < pressure) {
            maxPressure = pressure
            displayMaxPressure.text = "Maximum Pressure: " + maxPressure.toString()
        }
        if (maxSize < size) {
            maxSize = size
            displayMaxSize.text = "Maximum Size: " + maxSize.toString()
        }
    }
}
